From b24b74e563dc5cd6c817839dbc33f75009859ece Mon Sep 17 00:00:00 2001
From: tobitege <tobiasteschner@googlemail.com>
Date: Sun, 12 Apr 2015 15:24:29 +0200
Subject: [PATCH 32/47] Fix array exception in createAutoBrightnessSpline

Function does not check array lenghts and thus can cause bootloops
if brightness array configurations are invalid (different issue).

Change-Id: I8614aeb810c088b9d0274e54696856671bdcfde9
Signed-off-by: Adesh15 <adesikha15@gmail.com>
---
 .../com/android/server/display/DisplayPowerController.java   | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/display/DisplayPowerController.java b/services/core/java/com/android/server/display/DisplayPowerController.java
index 0be446d0c99..b41325f9e0b 100644
--- a/services/core/java/com/android/server/display/DisplayPowerController.java
+++ b/services/core/java/com/android/server/display/DisplayPowerController.java
@@ -1438,7 +1438,14 @@ final class DisplayPowerController implements AutomaticBrightnessController.Call
             return null;
         }
         try {
-            final int n = brightness.length;
+            int n = brightness.length;
+            final int m = lux.length;
+            if (n == 0 || m == 0) {
+              return null;
+            }
+            if (m < n) {
+              n = m;
+            }
             float[] x = new float[n];
             float[] y = new float[n];
             y[0] = normalizeAbsoluteBrightness(brightness[0]);
@@ -1458,6 +1465,9 @@ final class DisplayPowerController implements AutomaticBrightnessController.Call
         } catch (IllegalArgumentException ex) {
             Slog.e(TAG, "Could not create auto-brightness spline.", ex);
             return null;
+        } catch (ArrayIndexOutOfBoundsException ex) {
+            Slog.e(TAG, "Could not create auto-brightness spline (index fault).", ex);
+            return null;
         }
     }
 
-- 
2.11.0

